processor/blockprocessor/block_processor.go

```diff
@@ -3,146 +3,64 @@ package blockprocessor
 import (
 	"context"
 	"fmt"
+	"github.com/algorand/indexer/exporters"
 
 	log "github.com/sirupsen/logrus"
 
-	"github.com/algorand/go-algorand/config"
 	"github.com/algorand/go-algorand/data/basics"
 	"github.com/algorand/go-algorand/data/bookkeeping"
 	"github.com/algorand/go-algorand/data/transactions"
 	"github.com/algorand/go-algorand/ledger"
 	"github.com/algorand/go-algorand/ledger/ledgercore"
-	"github.com/algorand/go-algorand/rpcs"
-
 	"github.com/algorand/indexer/accounting"
 	"github.com/algorand/indexer/idb"
 	"github.com/algorand/indexer/processor"
 	indexerledger "github.com/algorand/indexer/processor/eval"
 	"github.com/algorand/indexer/util"
 )
 
-type blockProcessor struct {
-	handler func(block *ledgercore.ValidatedBlock) error
-	ledger  *ledger.Ledger
-	logger  *log.Logger
-}
-
-// MakeProcessorWithLedger creates a block processor with a given ledger
-func MakeProcessorWithLedger(logger *log.Logger, l *ledger.Ledger, handler func(block *ledgercore.ValidatedBlock) error) (processor.Processor, error) {
+// MakeProcessorWithLedger creates a Processor with a given ledger
+func MakeProcessorWithLedger(logger *log.Logger, l *ledger.Ledger, genericHandler interface{}) (processor.Processor, error) {
 	if l == nil {
 		return nil, fmt.Errorf("MakeProcessorWithLedger() err: local ledger not initialized")
 	}
-	err := addGenesisBlock(l, handler)
+	var proc processor.Processor
+	if exp, ok := genericHandler.(exporters.Exporter); ok {
+		proc = &ledgerExporterProcessor{logger: logger, ledger: l, exporter: exp}
+	} else if handler, ok := genericHandler.(func(block *ledgercore.ValidatedBlock) error); ok {
+		proc = &blockProcessor{logger: logger, ledger: l, handler: handler}
+	} else if genericHandler == nil {
```

Comment:
I think if genericHandler == nil the if statements before this one will fail with an error? I don't believe we are allowed to case nil to a type.